spec/models/green_flag/feature_spec.rb

```diff
@@ -132,4 +132,53 @@
       end
     end
   end
+
+  describe '#requires_manual_deletion?' do
+    let(:feature)           { GreenFlag::Feature.create(code: 'big_font') }
+    let(:feature_decisions) { [] }
+
+    context 'with more than 1,000,000 feature decisions' do
+      it 'returns true' do
+        expect(feature_decisions).to receive(:count).and_return(2_000_000)
+        expect(feature).to receive(:feature_decisions).and_return(feature_decisions)
+
+        expect(feature.requires_manual_deletion?).to be_true
+      end
+    end
+
+    context 'with fewer than 1,000,000 feature decisions' do
+      it 'returns false' do
+        expect(feature_decisions).to receive(:count).and_return(500_000)
+        expect(feature).to receive(:feature_decisions).and_return(feature_decisions)
+
+        expect(feature.requires_manual_deletion?).to be_false
+      end
+    end
+  end
+
+  describe 'delete_associated_data' do
+    let(:feature) { GreenFlag::Feature.create(code: 'big_font') }
+
+    before(:each) do
+      5.times do
+        FactoryGirl.create(:green_flag_rule, feature_id: feature.id)
+        FactoryGirl.create(:green_flag_feature_event, feature_id: feature.id)
+        FactoryGirl.create(:green_flag_feature_decision, feature_id: feature.id)
+      end
+
+      feature.delete_associated_data
+    end
+
+    it 'deletes all associated rules' do
+      expect(feature.rules.count).to eq 0
+    end
+
+    it 'deletes all associated events' do
+      expect(feature.feature_events.count).to eq 0
+    end
+
+    it 'deletes all associated feature decisions' do
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.